test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +114,80 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_exception_nested_in_json
```

Comment:
Yeah, in fact I don't even think we need an extra function. More like:

```
def test_ignore_nested_exception
  ...
  lines = [
    # Json payload.
    %({"timestamp": {"nanos": 998152494, "seconds": 1496420064}, "message": "#{single_line_exception}", "thread": 139658267147048, "severity": "ERROR"}\n),
    %({"timestamp": {"nanos": 5990266, "seconds": 1496420065}, "message": "next line", "thread": 139658267147048, "severity": "INFO"}\n)

    # Text payload.
    %(prefixed: #{single_line_exception}\n),
    %(next line\n)
  ]
  ...
end
```

The only difference between these two tests are the input:
```
14c14 // I left a separate comment for this. I think they don't necessarily need to be different here, which should make it pretty easy to combine these two tests.
<       single_line_exception = PYTHON_EXC.gsub("\n", "\\n")
---
>       single_line_exception = exception.gsub("\n", "\\n")
18,19c18,19
<         %({"timestamp": {"nanos": 998152494, "seconds": 1496420064}, "message": "#{single_line_exception}", "thread": 139658267147048, "severity": "ERROR"}\n),
<         %({"timestamp": {"nanos": 5990266, "seconds": 1496420065}, "message": "next line", "thread": 139658267147048, "severity": "INFO"}\n)
---
>         %(prefixed: #{single_line_exception}\n),
>         %(next line\n)
```